extractor.go

```diff
@@ -83,10 +83,10 @@ func (e extractors) getUnpackFunction(data []byte) unpackFunc {
 	return nil
 }
 
-// getUnpackFunctionByFileName identifies the correct extractor based on file extension.
-func (e extractors) getUnpackFunctionByFileName(ext string) unpackFunc {
+// GetUnpackFunctionByFileName identifies the correct extractor based on file extension.
+func (e extractors) GetUnpackFunctionByFileName(name string) unpackFunc {
 	// get file extension from file name
-	ext = strings.ToLower(ext)
+	ext := strings.ToLower(name)
 	if strings.Contains(ext, ".") {
```

Comment:
Hehe, i thought as well about it. In the current implementation, the file can be just the name of the extension , e.g., `name="tar"`, then `tar` would lead to a `return true`.